packages/retail-ui-extensions/src/extension-points/index.ts

```diff
@@ -1,10 +1,16 @@
-export type {ExtensionPoints, ExtensionPoint} from './extension-points';
-export type {RenderExtension} from './render-extension';
-export type {
-  RenderExtensions,
-  RenderExtensionPoint,
-  AllowedComponentsForRenderExtension,
-  ArgumentsForExtension,
-  ApiForRenderExtension,
-  ReturnTypeForExtension,
-} from './types';
+import {
+  SmartGridExtensionPoint,
+  SmartGridExtensionApi,
+  SmartGridExtensionPointCallback,
+} from './smart-grid-extension';
+
+export type {SmartGridExtensionPoint} from './smart-grid-extension';
+
+export type {ExtensionResult, RenderableExtensionCallback} from './types';
+
+// Generic public interfaces combining all POS extension points.
+export type ExtensionPoint = SmartGridExtensionPoint;
```

Comment:
These are basically just generic public interfaces. The user won't see "SmartGridExtensionPoint" but just ExtensionPoint. This also means in the future when we define more extension points like OrderExtensionPoint, it will be combined in this generic ExtensionPoint type